pkg/storaged/fsys-tab.jsx

```diff
@@ -18,24 +18,279 @@
+            else if (old_config && new_config && (new_dir != old_dir || new_opts != old_opts)) {
+                return block.UpdateConfigurationItem(new_config, old_config, {});
+            }
+        }
+
+        function maybe_unmount() {
+            if (block_fsys.MountPoints.length > 0)
+                return block_fsys.Unmount({ });
+            else
+                return Promise.resolve();
+        }
+
+        function maybe_mount() {
+            if (mode == "mount" || (mode == "update" && is_filesystem_mounted)) {
+                return (block_fsys.Mount({ })
+                        .catch(error => {
+                            return (undo()
+                                    .then(() => block_fsys.Mount({ }))
+                                    .then(() => Promise.reject(error))
+                                    .catch(ignored_error => {
+                                        console.warn("Error during undo:", ignored_error);
+                                        return Promise.reject(error);
+                                    }));
+                        }));
+            } else
+                return Promise.resolve();
+        }
+
+        // We need to reload systemd twice: Once at the beginning so
+        // that it is up to date with whatever is currently in fstab,
+        // and once at the end to make it see our changes.  Otherwise
+        // systemd might do some uexpected mounts/unmounts behind our
+        // backs.
+
+        return (utils.reload_systemd()
+                .then(maybe_unmount)
+                .then(() => {
+                    if (!old_config && new_config)
+                        return (block.AddConfigurationItem(new_config, {})
+                                .then(maybe_mount)
+                                .then(utils.reload_systemd));
+                    else if (old_config && !new_config)
+                        return block.RemoveConfigurationItem(old_config, {}).then(utils.reload_systemd);
+                    else if (old_config && new_config && (new_dir != old_dir || new_opts != old_opts))
+                        return (block.UpdateConfigurationItem(old_config, new_config, {})
+                                .then(maybe_mount)
+                                .then(utils.reload_systemd));
+                    else if (new_config && !is_mounted(client, block))
+                        return maybe_mount();
+                }));
+    }
+
+    function remove() {
+        dlg.run(null, maybe_update_config("", "").then(() => dlg.close()));
+    }
+
+    let fields = null;
+    if (mode == "mount" || mode == "update")
+        fields = [
+            TextInput("mount_point", _("Mount Point"),
+                      {
+                          value: old_dir,
+                          validate: function (val) {
+                              if (val === "")
+                                  return _("Mount point cannot be empty");
+                          }
```

Comment:
Maybe this should also ensure that it's an absolute path, i. e. it starts with '/'?

Reply:
You can actually use a relative path in fstab, and is treated as relative to "/".

But, yes, we should enforce a leading "/" in Cockpit (and add it if missing in fstab), I'd say.

Reply:
Interesting; it seems a little confusing indeed, as if someone writes "some/path" into fstab, and runs "mount some/path" from anywhere except cwd==/, that's bound to fail, right?

Reply:
I am not sure of the details, tbh.  What about silently adding a missing "/" upon "Apply"?  We already add a missing "/" when reading from fstab.

Reply:
@andreasn, any opinion?

Basically, we don't want people to use mount points that don't start with a "/".  The two options are:  Make it an error to enter a mount point without a starting "/".  Add the "/" silently if people don't write it out explicitly.

In any case, Cockpit will always show a mount points with a starting "/".

Reply:
I'd slightly prefer being explicit rather than change things behind the user's back. Either way, this is fine for follow-up material, no need to stall this PR on this detail IMHO.

Reply:
I see your PR just adds the `/`. So it essentially behaves like how systemd interprets fstab, i. e. always relative to `/`. +1 on going with that for now, and possibly changing it in a follow-up.